### PR TITLE
Add E2E Gardener upgrade test Prow jobs

### PR DESCRIPTION
**Pull request by seshachalam-yv**
/kind enhancement
**What this PR does / why we need it**:
This PR adds new Prow jobs for Gardener upgrade tests. 
Gardener upgrade tests are introduce as part of this PR https://github.com/gardener/gardener/pull/7122
 

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener/issues/2692 https://github.com/gardener/gardener/pull/7122

**Special notes for your reviewer**:

### CODE REVIEW

File: config/jobs/gardener/gardener-e2e-gardener-upgrade-kind-ha-single-zone.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    - release-v\d+.\d+ # don't run on release branches for now (add a job per branch later)
+    decorate: true
+    decoration_config:
+      timeout: 60m
+      grace_period: 15m
+    labels:
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    annotations:
+      description: Runs end-to-end tests on a Seed with single zone with a Shoot with failure tolerance type 'node' for gardener developments in pull requests
```
Comment (oliver-goetz) [inline config/jobs/gardener/gardener-e2e-gardener-upgrade-kind-ha-single-zone.yaml:17]:
```suggestion
      description: Runs end-to-end Gardener upgrade tests on a Seed with single zone with a Shoot with failure tolerance type 'node' for gardener developments in pull requests
```

File: config/jobs/gardener/gardener-e2e-gardener-upgrade-kind-ha-single-zone.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    base_ref: master
+  decorate: true
+  decoration_config:
+    timeout: 60m
+    grace_period: 15m
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  annotations:
+    description: Runs end-to-end tests on a Seed with single zone with a Shoot with failure tolerance type 'node' for gardener developments periodically
```
Comment (oliver-goetz) [inline config/jobs/gardener/gardener-e2e-gardener-upgrade-kind-ha-single-zone.yaml:55]:
```suggestion
    description: Runs end-to-end Gardener upgrade tests on a Seed with single zone with a Shoot with failure tolerance type 'node' for gardener developments periodically
```

**Review by oliver-goetz — CHANGES_REQUESTED** [review]
Please harmonize the test names and the filenames accordingly.
The make targets and descriptions of single zone tests look wrong too.

**Comment by seshachalam-yv** [comment]
> Please harmonize the test names and the filenames accordingly. The make targets and descriptions of single zone tests look wrong too.

Thanks for review @oliver-goetz 👍 
Adapted names based on new [g/g make targets](https://github.com/gardener/gardener/pull/7122/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R415-R421)

- `ci-e2e-kind-upgrade`
- `ci-e2e-kind-ha-single-zone-upgrade`
- `ci-e2e-kind-ha-multi-zone-upgrade`

**Review by oliver-goetz — APPROVED** [review]
/lgtm

**Review by oliver-goetz — COMMENTED** [review]
Found one more thing 🕵️‍♂️ 
Please make all jobs presubmit jobs optional.

/remove-lgtm

**Pull request merged**